File: patterns/associated-items/rules-js/associated-items-group.js
const { rules, triggers, items } = require('openhab');

rules.JSRule({
  name: "Zugehoeriges Item ueber Gruppenzugehoerigkeit finden",
  triggers: [triggers.ItemGroupStateChangeTrigger("gSensors")], // Angepasst
  execute: (event) => {
    const sensors = items.getItem("gSensors").members; //[cite: 3]

    const status1 = sensors.find((i) => i.name === event.itemName + "_Status"); //[cite: 3]
    const status2 = sensors.find((i) => i.tags.includes("Status")); //[cite: 3]

    console.log("status1=" + (status1 && status1.name) + " status2=" + (status2 && status2.name)); //[cite: 3]
  }
});

File: patterns/associated-items/rules-python/associated_items_group.py
from openhab import rule, Registry
from openhab.triggers import GroupStateChangeTrigger


@rule(triggers=[GroupStateChangeTrigger("gSensors")])
class ZugehoerigesItemUeberGruppenzugehoerigkeitFinden:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        item_name = event.getItemName()
        sensors = Registry.getItem("gSensors").getAllMembers()

        # Ansatz: Item-Name
        status1 = next((i for i in sensors if i.getName() == item_name + "_Status"), None)

        # Ansatz: Item-Tag
        status2 = next((i for i in sensors if "Status" in i.getTags()), None)

        # Ansatz: Item-Tags
        status3 = next(
            (i for i in sensors if all(t in i.getTags() for t in ("Status", "Power"))),
            None,
        )

        # Ansatz: mehrere Kriterien
        status4 = next(
            (i for i in sensors
             if "Status" in i.getTags() and i.getName().endswith("_Status") and i.getType() == "Switch"),
            None,
        )

        self.logger.info(
            "status1={} status2={}".format(
                status1.getName() if status1 else None,
                status2.getName() if status2 else None,
            )
        )
